### ISSUE TITLE
[7.4.0] Crash during exec log generation

### ISSUE THREAD

**Issue by fmeum** [bazelbuild/bazel#23884]
### Description of the bug:

```
-- MODULE.bazel --
-- pkg/BUILD --
DATA = glob(["*.txt"])

sh_binary(
    name = "tool",	
    srcs = ["tool.sh"],
    data = DATA,
    args = ["$(rootpath {})".format(file) for file in DATA],
)
genrule(
    name = "gen1",
    outs = ["out1"],
    cmd = "$(location :tool) > $@",
    tools = [":tool"],
)
genrule(
    name = "gen2",
    outs = ["out2"],
    cmd = """
$(location :tool) > $@
$(location :tool) >> $@
""",
    tools = [":tool"],
)
-- pkg/tool.sh --
#!/bin/bash
cat "$@"
-- pkg/file1.txt --
old
-- pkg/file2.txt --
unchanged
```

results in:

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ActionLookupData0{actionLookupKey=ConfiguredTargetKey{label=//pkg:gen1, config=BuildConfigurationKey[4cf1c944324811e840e4f5049a0d662eb35b2ebf65bdb7fdb6de7d17810fb043]}, actionIndex=0}' (requested by nodes 'TargetCompletionKey{topLevelArtifactContext=com.google.devtools.build.lib.analysis.TopLevelArtifactContext@90904c3b, actionLookupKey=ConfiguredTargetKey{label=//pkg:gen1, config=BuildConfigurationKey[4cf1c944324811e840e4f5049a0d662eb35b2ebf65bdb7fdb6de7d17810fb043]}, willTest=false}')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:550)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:414)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "com.google.devtools.build.lib.actions.RunfilesSupplier$RunfilesTree.getExecPath()" because "runfilesTree" is null
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logRunfilesTree(CompactSpawnLogContext.java:561)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.lambda$logInputSet$3(CompactSpawnLogContext.java:409)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logEntry(CompactSpawnLogContext.java:712)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logInputSet(CompactSpawnLogContext.java:385)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.lambda$logInputSet$3(CompactSpawnLogContext.java:394)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logEntry(CompactSpawnLogContext.java:695)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logInputSet(CompactSpawnLogContext.java:385)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logInputs(CompactSpawnLogContext.java:328)
	at com.google.devtools.build.lib.exec.CompactSpawnLogContext.logSpawn(CompactSpawnLogContext.java:229)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:191)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:118)
	at com.google.devtools.build.lib.exec.SpawnStrategyResolver.exec(SpawnStrategyResolver.java:45)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.execute(SpawnAction.java:263)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.executeAction(SkyframeActionExecutor.java:1159)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1076)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:165)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:94)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:573)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:862)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:334)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:172)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:461)
	... 7 more
```

### Which category does this issue belong to?

Performance

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Comment by iancha1992** [comment]
@fmeum doesn't this have to be included in 8.0.0 as well?

**Comment by iancha1992** [comment]
@bazel-io fork 8.0.0

**Comment by fmeum** [comment]
@iancha1992 No, it's a bug in the cherry-pick, it doesn't affect Bazel 8.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
